pkg/api/health.go

```diff
@@ -182,35 +182,39 @@ func calculateJobResultStatistics(results []apitype.Job) (currStats, prevStats s
 		prevPercentages = append(prevPercentages, result.PreviousPassPercentage)
 	}
 
-	data := stats.LoadRawData(currPercentages)
-	mean, _ := stats.Mean(data)
-	sd, _ := stats.StandardDeviation(data)
-	quartiles, _ := stats.Quartile(data)
-	p95, _ := stats.Percentile(data, 95)
-
-	currStats.Mean = mean
-	currStats.StandardDeviation = sd
-	currStats.Quartiles = []float64{
-		testreportconversion.ConvertNaNToZero(quartiles.Q1),
-		testreportconversion.ConvertNaNToZero(quartiles.Q2),
-		testreportconversion.ConvertNaNToZero(quartiles.Q3),
-	}
-	currStats.P95 = p95
-
-	data = stats.LoadRawData(prevPercentages)
-	mean, _ = stats.Mean(data)
-	sd, _ = stats.StandardDeviation(data)
-	quartiles, _ = stats.Quartile(data)
-	p95, _ = stats.Percentile(data, 95)
-
-	prevStats.Mean = mean
-	prevStats.StandardDeviation = sd
-	prevStats.Quartiles = []float64{
-		testreportconversion.ConvertNaNToZero(quartiles.Q1),
-		testreportconversion.ConvertNaNToZero(quartiles.Q2),
-		testreportconversion.ConvertNaNToZero(quartiles.Q3),
-	}
-	prevStats.P95 = p95
```

Comment:
Not related to the work in this PR but we are getting back empty percentages arrays that generate NaN mean, etc. which then cause JSON serialization errors in the e2e so added a check to make sure we have data.

Reply:
Looks like https://github.com/openshift/sippy/pull/844/files#diff-1094b7bba124eec28f852d0ca61f46517f446298e4e4d119f5f6c771e1032c0bL157 

Removed some code that was handling the NaNs previously

`Percentage: testreportconversion.ConvertNaNToZero(testReport.CurrentPassPercentage),`

Reply:
Not sure that code is in play, it is looking more like the currentRuns is < 7 so we continue in calculateJobResultStatistics loop and wind up with no entries in currPercentages or prevPercentages. 

[PR 854](https://github.com/openshift/sippy/pull/854) was used to validate the issue was external to this PR